### PR TITLE
a8n: Handle GitHub review events when reviewer is team

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes https://github.com/sourcegraph/sourcegraph/issues/7715 by loading the reviewer even if it is a team instead of an actor. Previously the fields were blank because the GraphQL didn't return an `Actor` in the `requestedReviewer` field.

That caused multiple `ChangesetEvents` that wrap different `github.ReviewRequestedEvent`s to have the same key, because their `RequestedReviewer` was empty.

To fix that this PR loads the `requestedTeam` and sets it on the `github.ReviewRequestedEvent` and `github.ReviewRequestRemovedEvent` if it is set instead of an actor.

### CODE REVIEW

File: internal/a8n/types.go [hunk 1]
```diff
@@ -848,6 +848,10 @@ func (e *ChangesetEvent) Update(o *ChangesetEvent) {
 			e.RequestedReviewer = o.RequestedReviewer
 		}
 
+		if e.RequestedTeam == (github.Team{}) {
```
Comment (ryanslade) [inline internal/a8n/types.go]: Does it need to be surrounded by parentheses?

File: internal/a8n/types.go [hunk 2]
```diff
@@ -863,6 +867,10 @@ func (e *ChangesetEvent) Update(o *ChangesetEvent) {
...
```
Comment (ryanslade) [inline internal/a8n/types.go]: Does it need to be surrounded by parentheses?

File: internal/a8n/types.go (hunk 1 above)
Comment (mrnugget) [reply internal/a8n/types.go]: Yep, otherwise it's a syntax error.

File: internal/a8n/types.go (hunk 2 above)
Comment (mrnugget) [reply internal/a8n/types.go]: Yes, otherwise it's a syntax error.

Comment (ryanslade) [reply internal/a8n/types.go]: Interesting, it's only an error across packages. Within a package it works fine.

**Pull request merged**